graylog2-web-interface/src/views/components/visualizations/worldmap/WorldMapVisualization.jsx

```diff
@@ -1,37 +1,48 @@
 // @flow strict
 import React, { useContext } from 'react';
 import PropTypes from 'prop-types';
-import { flow, fromPairs, get, zip } from 'lodash';
+import { flow, fromPairs, get, zip, isEmpty } from 'lodash';
 
 import { AggregationType } from 'views/components/aggregationbuilder/AggregationBuilderPropTypes';
 import WorldMapVisualizationConfig from 'views/logic/aggregationbuilder/visualizations/WorldMapVisualizationConfig';
 import Viewport from 'views/logic/aggregationbuilder/visualizations/Viewport';
 import type { VisualizationComponent, VisualizationComponentProps } from 'views/components/aggregationbuilder/AggregationBuilder';
+import type { Rows } from 'views/logic/searchtypes/pivot/PivotHandler';
 
 import MapVisualization from './MapVisualization';
-import { extractSeries, formatSeries } from '../ChartData';
+import { extractSeries, formatSeries, getLeafsFromRows, getXLabelsFromLeafs } from '../ChartData';
 import transformKeys from '../TransformKeys';
 import RenderCompletionCallback from '../../widgets/RenderCompletionCallback';
 
-const arrayToMap = ([name, x, y]) => ({ name, x, y });
-const lastKey = keys => keys[keys.length - 1];
-const mergeObject = (prev, last) => Object.assign({}, prev, last);
+const _arrayToMap = ([name, x, y]) => ({ name, x, y });
+const _lastKey = keys => keys[keys.length - 1];
+const _mergeObject = (prev, last) => Object.assign({}, prev, last);
+const _createSeriesWithoutMetric = (rows: Rows) => {
+  const leafs = getLeafsFromRows(rows);
+  const xLabels = getXLabelsFromLeafs(leafs);
+  return { valuesBySeries: { 'No metric defined': xLabels.map(() => null) }, xLabels };
+};
 
 const WorldMapVisualization: VisualizationComponent = ({ config, data, editing, onChange, width, ...rest }: VisualizationComponentProps) => {
-  const onRenderComplete = useContext(RenderCompletionCallback);
   const { rowPivots } = config;
+  const onRenderComplete = useContext(RenderCompletionCallback);
+  const hasMetric = !isEmpty(config._value.series);
```

Comment:
`_value` is an internal field of the `AggregationWidgetConfig` class. please use `config.series` instead.